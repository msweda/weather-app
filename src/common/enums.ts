export enum RootRoute {
  Home = '/',
  Weather = '/weather',
  Forecast = '/forecast',
}

export enum AsyncStatus {
  Idle = 'IDLE',
  InProgress = 'IN_PROGRESS',
  Succeeded = 'SUCCEEDED',
  Failed = 'FAILED',
}
