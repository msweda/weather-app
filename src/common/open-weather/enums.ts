export enum UnitType {
  Scientific = 'scientific',
  Metric = 'metric',
  Imperial = 'imperial',
}
