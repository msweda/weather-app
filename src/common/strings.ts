export const appHeading = 'Matthew Sweda\'s Weather App';
export const phoneNumber = '(248)-770-7487';
export const email = 'msweda.developer@gmail.com';
export const location = 'Detroit, MI';
export const copyright = (timestamp: number) => `Â© Matthew Sweda ${new Date(timestamp).getFullYear()}`;
export const emailLink = `mailto:${email}`;
export const linkedInLink = 'https://www.linkedin.com/in/matthew-sweda-118570ba/';
export const instagramLink = 'https://www.instagram.com/mattsweda/';