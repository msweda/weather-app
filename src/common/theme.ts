// create palette here: https://material.io/color/

import { createMuiTheme } from '@material-ui/core/styles'

export default createMuiTheme({
  typography: {
    useNextVariants: true,
  },
  palette: {
    primary: {
      main: '#2d4671',
    },
    secondary: {
      main: '#6ba099',
    },
    background: {
      default: '#2d4671',
      paper: '#eee',
    },
    text: {
      primary: '#fff',
      secondary: '#000',
    },
  },
  overrides: {
    MuiCssBaseline: {
      '@global': {
        body: {
          display: 'flex',
          minHeight: '100vh',
        },
        '#app': {
          flex: 1,
          display: 'flex',
          flexDirection: 'column',
        },
      },
    },
  },
})
