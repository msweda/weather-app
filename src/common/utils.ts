import createBrowserHistory from 'history/createBrowserHistory'

export const browserHistory = createBrowserHistory()

export function isGeolocationAvailable(): boolean {
  return 'geolocation' in navigator
}

export function getCurrentPosition() {
  return new Promise<Position>((resolve, reject) => {
    if (!isGeolocationAvailable()) {
      throw new Error('Geolocation is unavailable')
    }
    navigator.geolocation.getCurrentPosition(
      position => resolve(position),
      error => reject(error)
    )
  })
}
