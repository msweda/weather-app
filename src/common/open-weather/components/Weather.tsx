import { createStyles, WithStyles, withStyles } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'
import * as React from 'react'
import { Weather as WeatherInterface } from '../interfaces'
import { weatherIconUrl } from '../strings'

class Weather extends React.Component<Props> {
  render() {
    const { classes, weather } = this.props
    return (
      <div className={classes.weatherIcons}>
        {weather.map((element, index) => (
          <div
            key={`weather-array-element-${index}`}
            className={classes.weatherArrayElement}
          >
            <img src={weatherIconUrl(element.icon)} width="30" height="30" />
            <Typography color="primary">{element.main}</Typography>
          </div>
        ))}
      </div>
    )
  }
}

const styles = () =>
  createStyles({
    weatherIcons: {
      display: 'flex',
      flexDirection: 'row',
      justifyContent: 'center',
      alignItems: 'center',
    },
    weatherArrayElement: {
      margin: 10,
    },
  })

interface Props extends WithStyles<typeof styles> {
  weather: WeatherInterface
}

export default withStyles(styles)(Weather)
