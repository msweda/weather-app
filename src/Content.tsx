import { RootRoute } from '@/common/enums'
import Forecast from '@/forecast/index'
import Weather from '@/weather/index'
import {
  createStyles,
  Theme,
  WithStyles,
  withStyles,
} from '@material-ui/core/styles'
import * as React from 'react'
import { Redirect, Route, Switch } from 'react-router-dom'

const Content: React.FC<Props> = ({ classes }) => (
  <div className={classes.content}>
    <Switch>
      <Route exact={true} path={RootRoute.Weather} component={Weather} />
      <Route exact={true} path={RootRoute.Forecast} component={Forecast} />
      <Route component={DefaultContent} />
    </Switch>
  </div>
)

const DefaultContent: React.FC<any> = () => <Redirect to={RootRoute.Weather} />

const styles = (theme: Theme) =>
  createStyles({
    content: {
      flex: 1,
      display: 'flex',
      flexDirection: 'column',
      backgroundColor: theme.palette.background.paper,
    },
  })

interface Props extends WithStyles<typeof styles> {}

export default withStyles(styles)(Content)
