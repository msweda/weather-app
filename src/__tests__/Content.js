import { RootRoute } from '@/common/enums'
import Forecast from '@/forecast/index'
import Weather from '@/weather/index'
import { createMount, createShallow } from '@material-ui/core/test-utils'
import * as React from 'react'
import { MemoryRouter } from 'react-router-dom'
import Content from '../Content'

describe('<Content />', () => {
  it('should match snapshot', () => {
    const shallow = createShallow({ untilSelector: 'Content' })
    const wrapper = shallow(<Content />)
    expect(wrapper).toMatchSnapshot()
  })

  describe('routing', () => {
    let mount

    beforeEach(() => {
      mount = createMount()
    })

    afterEach(() => {
      mount.cleanUp()
    })

    it('should render <Weather /> for the appropriate url', () => {
      const wrapper = mount(
        <MemoryRouter initialEntries={[RootRoute.Weather]}>
          <Content />
        </MemoryRouter>
      )
      expect(wrapper.find(Weather).exists()).toBeTruthy()
      expect(wrapper.find(Forecast).exists()).toBeFalsy()
    })

    it('should render the <Forecast /> component for the appropriate url', () => {
      const wrapper = mount(
        <MemoryRouter initialEntries={[RootRoute.Forecast]}>
          <Content />
        </MemoryRouter>
      )
      expect(wrapper.find(Weather).exists()).toBeFalsy()
      expect(wrapper.find(Forecast).exists()).toBeTruthy()
    })

    it('should render the <Weather /> component when the url matches no known route', () => {
      const wrapper = mount(
        <MemoryRouter initialEntries={['/some-other-route']}>
          <Content />
        </MemoryRouter>
      )
      expect(wrapper.find(Weather).exists()).toBeTruthy()
      expect(wrapper.find(Forecast).exists()).toBeFalsy()
    })
  })
})
