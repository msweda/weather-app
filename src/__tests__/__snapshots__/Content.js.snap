// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content /> should match snapshot 1`] = `
<div
  className="Content-content-1"
>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/weather"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/forecast"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</div>
`;
